internal/sdkv2provider/resource_cloudflare_device_dex_tests_test.go

```diff
@@ -0,0 +1,108 @@
+package sdkv2provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccCloudflareDeviceDexTests(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_device_dex_tests.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheckAccount(t)
+		},
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareDeviceDexTestsHttp(accountID, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "account_id", accountID),
+					resource.TestCheckResourceAttr(name, "name", rnd),
+					resource.TestCheckResourceAttr(name, "description", rnd),
+					resource.TestCheckResourceAttr(name, "interval", "0h30m0s"),
+					resource.TestCheckResourceAttr(name, "enabled", "true"),
+					resource.TestCheckResourceAttr(name, "data.0.host", "https://dash.cloudflare.com/home"),
+					resource.TestCheckResourceAttr(name, "data.0.kind", "http"),
+					resource.TestCheckResourceAttr(name, "data.0.method", "GET"),
+				),
+			},
+			{
+				Config: testAccCloudflareDeviceDexTestsTraceroute(accountID, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "account_id", accountID),
+					resource.TestCheckResourceAttr(name, "name", rnd),
+					resource.TestCheckResourceAttr(name, "description", rnd),
+					resource.TestCheckResourceAttr(name, "interval", "0h30m0s"),
+					resource.TestCheckResourceAttr(name, "enabled", "true"),
+					resource.TestCheckResourceAttr(name, "data.0.host", "dash.cloudflare.com"),
+					resource.TestCheckResourceAttr(name, "data.0.kind", "http"),
+				),
+			},
+			{
+				Config: testAccCloudflareDeviceDexTestsTracerouteIpv4(accountID, rnd),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "account_id", accountID),
+					resource.TestCheckResourceAttr(name, "name", rnd),
+					resource.TestCheckResourceAttr(name, "description", rnd),
+					resource.TestCheckResourceAttr(name, "interval", "0h30m0s"),
+					resource.TestCheckResourceAttr(name, "enabled", "true"),
+					resource.TestCheckResourceAttr(name, "data.0.host", "1.1.1.1"),
+					resource.TestCheckResourceAttr(name, "data.0.kind", "http"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCloudflareDeviceDexTestsHttp(accountID, rnd string) string {
+	return fmt.Sprintf(`
+	resource "cloudflare_device_dex_tests" "%[1]s" {
+		test_id = "%[2]s"
```

Comment:
`test_id` doesn't appear in your schema. is this the value we are using for `id`? if so, we don't need it here.

Reply:
Yes for this resource we used `test_id` to represent what would usually be the `id` field. Thanks for the heads up.